### ISSUE TITLE
http proxy doesn't close connection after sending 'Could not connect'

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10843]
|<img alt="jeffsilver's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| jeffsilver reported|
|-|-|
|Trac ID|trac#1089|
|Type|defect|
|Created|2005-06-20 16:18:20Z|

Attachments:

* [1089.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/785/785a7925363bf133a7c5413c563f331c5e02cc69/d7410873dfd8f0a0e435057192fda3e7020f6a23.diff) (918 bytes) - added by therve on 2007-11-22 11:22:24Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__1089 1089
type__defect defect
reporter__jeffsilver jeffsilver
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1119284300000000 1119284300000000
changetime__1198843146000000 1198843146000000
version__ 
owner__therve therve
cc__jknight cc__jeffsilver
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="jeffsilver's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|jeffsilver commented|
|-|-|

```
#!html
<pre>
In web/proxy.py, ProxyClientFactory.clientConnectionFailed(), an error page is
sent to the client, but the connection is not closed.
I think this can be fixed by inserting:
    self.father.transport.loseConnection()
at the end of the method (but I'm not certain that this is clean).
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="jeffsilver's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|jeffsilver commented|
|-|-|

```
#!html
<pre>
I've included this fix in my patch for #1149
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

[#2922](https://github.com/twisted/twisted/issues/2922) has been closed as a duplicate of this.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve removed owner|
|-|-|

I think that doesn't deserver a branch :). Ready to review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @therve|
|-|-|

Beautiful, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set status to `closed`|
|-|-|

(In [22159]) Close connection twisted.web.proxy.ProxyClientFactory when reporting a 501
error.

Author: therve
Reviewer: exarkun
Fixes [#10843](https://github.com/twisted/twisted/issues/10843)
